### PR TITLE
Feat: Optimize reportResult's required gas

### PR DESCRIPTION
**Pull request by guidiaz**

### CODE REVIEW

File: test/wrb.js [hunk 1]
```diff
@@ -359,20 +359,6 @@ contract("WitnetRequestBoard", ([
...
-
-      // Should revert when reading the request since the bytecode has changed
-      await expectRevert(
-        this.WitnetRequestBoard.readRequestBytecode(queryId, {
-          from: other,
-          gasPrice: 1,
-        }),
-        "bytecode changed after posting"
-      )
-    })
```
Comment (aesedepece) [inline test/wrb.js:375]: Why is this being removed?

File: contracts/impls/trustable/WitnetRequestBoardTrustableBase.sol [hunk 2]
```diff
@@ -694,16 +694,18 @@ abstract contract WitnetRequestBoardTrustableBase
...
+        __response.timestamp = _timestamp;
+        __response.drTxHash = _drTxHash;
+        __response.reporter = msg.sender;
+        __response.cborBytes = _cborBytes;
 
-        _safeTransferTo(payable(msg.sender), _query.request.reward);
+        _safeTransferTo(payable(msg.sender), __query.request.reward);
         emit PostedResult(_queryId, msg.sender);
+
+        delete __query.request;
```
Comment (aesedepece) [inline contracts/impls/trustable/WitnetRequestBoardTrustableBase.sol:709]: This is the core change of this PR. Am I understanding correctly that from now, in every single case, the `request` part of queries will be dropped as soon as we have a result reported?

This is interesting. I can't really think of an use case at this point for archiving the requests. I'm very supportive of this idea.

File: test/wrb.js (hunk 1 above)
Comment (guidiaz) [reply test/wrb.js:375]: Given a queryId, it doesn't make too much sense to check whether its request contract changed its bytecode after a response was already reported. 

The only place in which this check makes sense would be when calling `WRB.readRequestBytecode(queryId)` before a query gets solved (as if it was already solved, it should revert with "not in posted status"). So:
- this test will be moved to a more appropiate place
- implementation of `readRequestBytecode` needs to be revisited

In other words, I believe it should not matter at all if a request contract modifies its bytecode at any time after being posted to the WRB, as it will be certain that any eventual response will correspond to the bytecode the request had at the moment the query got posted. This consistency check was done in a previous version of `reportResult`, but not any more.

File: contracts/impls/trustable/WitnetRequestBoardTrustableBase.sol (hunk 2 above)
Comment (guidiaz) [reply contracts/impls/trustable/WitnetRequestBoardTrustableBase.sol:709]: Great. However, some more changes to the data model will be proposed as to prevent this PR from becoming a breaking change (and therefore, forcing to either deploy new proxy address, or making 100% sure there are no pending requests in the WRB before upgrading it)

Comment (aesedepece) [reply contracts/impls/trustable/WitnetRequestBoardTrustableBase.sol:696]: That sounds like music to my ears! :heart_eyes:

Comment (guidiaz) [reply contracts/impls/trustable/WitnetRequestBoardTrustableBase.sol:696]: Done in https://github.com/witnet/witnet-solidity-bridge/pull/216/commits/f228c5953a91084d73c8e580ccae815f7741f9ad.

**Review by aesedepece — APPROVED** [review]

**Review by aesedepece — APPROVED** [review]
